vck/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/Validator.kt

```diff
@@ -404,24 +425,37 @@ class Validator(
         val isRevoked = checkRevocationStatus(sdJwt) == RevocationStatus.REVOKED
         if (isRevoked)
             Napier.d("verifySdJwt: revoked")
-        // it's important to read again from source string to prevent different formats in serialization
-        val disclosureInputs = sdJwtSigned.rawDisclosures.map { it.hashDisclosure() }
-        disclosureInputs.forEach { discInput ->
-            if (sdJwt.disclosureDigests?.contains(discInput) != true) {
+
+        /** Map of serialized disclosure item (as [String]) to parsed item (as [SelectiveDisclosureItem]) */
+        val validDisclosures: Map<String, SelectiveDisclosureItem> = sdJwtSigned.rawDisclosures.filter {
+            // it's important to read again from source string to prevent different formats in serialization
```

Comment:
I can't really wrap my head around it. Could you explain it in a bit more detail, please?

Reply:
Discussed offline, see https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-13.html#name-disclosure-format-considera